sources/web/datalab/polymer/modules/ApiManager.ts

```diff
@@ -124,13 +124,33 @@ class ApiManager {
   static readonly terminalApiUrl = '/api/terminals';
 
   /**
+   * URL for retrieving the base path
+   */
+  static readonly basepathApiUrl = '/api/basepath';
+
+  /**
+   * URL for starting terminals, undefined until we call basepathApiUrl
+   */
+  static _basepath = '';
```

Comment:
This is unnecessary now, everyone should use the promise, which will return immediately if it already had the value resolved.